src/scores/continuous/murphy_impl.py

```diff
@@ -116,7 +117,8 @@ def murphy_score(
     for source, name in zip(sources, names):
         source.name = name
     result = xr.merge(sources)
-    reduce_dims = gather_dimensions(fcst.dims, obs.dims, reduce_dims, preserve_dims)
+    # Pylance doesn't list fcst.dims being matched to a list of strings, but this is well-tested and is fine
+    reduce_dims = gather_dimensions(fcst.dims, obs.dims, reduce_dims, preserve_dims) # type: ignore
```

Comment:
I'm going to put my foot down here and go against this since this problem likely will keep repeating itself all over the code base. 
Please add `Iterable[Hashable]` to the `fcst_dims` and `obs_dims` type hints in `utils.gather_dimensions`. 
https://docs.xarray.dev/en/stable/generated/xarray.DataArray.dims.html

PS, I can commit these changes as I've been testing them out locally as I review too if your pressed for time today.